File: WebApp/Models/Project.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace WebApi.Models
{
    public class ProjectDTO
    {
        public Guid Id { get; set; }

        public string Number { get; set; }

        public string Name { get; set; }

        public DateTime? StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        public bool AllowView { get; set; }

        public bool AllowAdd { get; set; }

        public bool AllowEdit { get; set; }

        public bool AllowDelete { get; set; }
    }
}

File: WebApp/wwwroot/scripts/domain/project.ts
module App.Models {
    "use strict";

    export interface IProject extends ng.resource.IResource<IProject> {
        id: string;
        number: string;
        name: string;
        startDate: Date;
        endDate: Date;
        notes: string;
        archived: boolean;
        organisation: Models.IOrganisation;
        createdBy: Models.IOrgUser;
        allowView: boolean;
        allowAdd: boolean;
        allowEdit: boolean;
        allowDelete: boolean;
        allowExportPdf: boolean;
        allowExportZip: boolean;
        lastEntry?: Date;
        assignmentsCount: number;
        teamsCount: number;
        isAggregate: boolean;

        isSelected: boolean;
    }

    export interface IProjectAssignment {
        orgUserId: string;
        orgUserName: string;
        isRootUser: boolean;
        isOwner: boolean;
        projectId: string;
        canAdd: boolean;
        canEdit: boolean;
        canView: boolean;
        canDelete: boolean;
        canExportPdf: boolean;
        canExportZip: boolean;
    }
}